pkg/revision/references.go

```diff
@@ -0,0 +1,58 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package revision
+
+import (
+	"github.com/istio-ecosystem/sail-operator/api/v1alpha1"
+	"github.com/istio-ecosystem/sail-operator/pkg/constants"
+)
+
+func GetReferencedRevisionFromNamespace(labels map[string]string) string {
+	// istio-injection label takes precedence over istio.io/rev
+	if labels[constants.IstioInjectionLabel] == constants.IstioInjectionEnabledValue {
+		return v1alpha1.DefaultRevision
+	}
+	revision := labels[constants.IstioRevLabel]
+	if revision != "" {
+		return revision
+	}
+	// TODO: if .Values.sidecarInjectorWebhook.enableNamespacesByDefault is true, then all namespaces except system namespaces should use the "default" revision
+
+	return ""
+}
+
+func GetReferencedRevisionFromPod(podLabels map[string]string) string {
+	// we only look at pod labels to identify injection intent
+	if podLabels[constants.IstioSidecarInjectLabel] != "false" {
+		if revisionFromPod := podLabels[constants.IstioRevLabel]; revisionFromPod != "" {
+			return revisionFromPod
+		}
```

Comment:
Nit. Since the scope of the variable is smaller, we can shorten its name. But yeah, it could have been shorter even in the previous code.
```suggestion
		if rev := podLabels[constants.IstioRevLabel]; rev != "" {
			return rev
		}
```